src/controllers/__tests__/auth.test.js

```diff
@@ -97,13 +191,121 @@ describe('Auth Endpoints', () => {
       expect(res.text).toMatch(/username already used/i);
     });
 
+    test('Should throw an error if the username is empty', async () => {
+      const res = await request(server)
+        .post('/api/auth/signup')
+        .send(userWithEmptyUsername);
+      expect(res.header.location).toBe(undefined);
+      expect(res.statusCode).toBe(422);
+      expect(res.text).toMatch(
+        '{"errors":[{"username":"Username should not be empty"}]}'
+      );
+    });
+
+    test('Should throw an error if the username is less than 3 characters', async () => {
+      const res = await request(server)
+        .post('/api/auth/signup')
+        .send(userWithNotValidUsername1);
+      expect(res.header.location).toBe(undefined);
+      expect(res.statusCode).toBe(422);
+      expect(res.text).toMatch(
+        '{"errors":[{"username":"Username must be more than three and less than 30 characters"}]}'
+      );
+    });
+
+    test('Should throw an error if the username contains special characters other than period and underscore', async () => {
+      const res = await request(server)
+        .post('/api/auth/signup')
+        .send(userWithNotValidUsername2);
+      expect(res.header.location).toBe(undefined);
+      expect(res.statusCode).toBe(422);
+      expect(res.text).toMatch(
+        '{"errors":[{"username":"Username can contain only alphanumeric characters, period and underscore"}]}'
+      );
+    });
+
```

Comment:
After adding a custom validation(see below), you can add a test case for username with spaces between letters like "chand bing".